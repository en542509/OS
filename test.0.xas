    std
    xor r0, r0
    loadi 1024, r1

loop:
    inc r0
    equ r1, r0
    br _end
    jr loop
_end:
    loadi 0, r1
