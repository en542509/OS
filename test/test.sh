#!/bin/bash

PREFIX=test
TESTS='0 1 2 3 4'
RAM='64 32 16 8 4 2'
PROG=runme.sh
count=0

if [ ! -x $PROG ]; then
  echo Error: $PROG is not in the current directory or is not executable
  exit 1
fi

for T in $TESTS; do
  for R in $RAM; do
    if [ -f $PREFIX.$T.xo ]; then  
      echo ===========================================
      echo Test file: $PREFIX.$T.xo  RAM: $R
      ./$PROG 0 $PREFIX.$T.xo 1 $R > $PREFIX.$T.out
      if diff $PREFIX.$T.out $PREFIX.$T.gold > /dev/null; then
        echo " " PASSED
        let "count = count + 1"
      else
        echo " " FAILED
      fi
      # rm $PREFIX.$T.out
    else
      echo Oops: file $PREFIX.$T.xo does not exist!
    fi
  done
done

echo =============================================
echo $count tests passed
