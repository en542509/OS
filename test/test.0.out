00> PC: 0002, State: 0002
00> Registers: 
00>      r00:0000 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 42, r0

00> PC: 0006, State: 0002
00> Registers: 
00>      r00:002a r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 0, r1

00> PC: 000a, State: 0002
00> Registers: 
00>      r00:002a r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 2, r3

00> PC: 000e, State: 0002
00> Registers: 
00>      r00:002a r01:0000 r02:0000 r03:0002 r04:0000 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: loadi 4, r4

00> PC: 0012, State: 0002
00> Registers: 
00>      r00:002a r01:0000 r02:0000 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: load r0, r2

00> PC: 0014, State: 0002
00> Registers: 
00>      r00:002a r01:0000 r02:0000 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: equ r2, r1

00> PC: 0016, State: 0003
00> Registers: 
00>      r00:002a r01:0000 r02:0000 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: br 4

00> PC: 001a, State: 0003
00> Registers: 
00>      r00:002a r01:0000 r02:0000 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: inc r1

00> PC: 001c, State: 0003
00> Registers: 
00>      r00:002a r01:0001 r02:0000 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: equ r1, r4

00> PC: 001e, State: 0002
00> Registers: 
00>      r00:002a r01:0001 r02:0000 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: br 10

00> PC: 0020, State: 0002
00> Registers: 
00>      r00:002a r01:0001 r02:0000 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: add r3, r0

00> PC: 0022, State: 0002
00> Registers: 
00>      r00:002c r01:0001 r02:0000 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: jr 240

00> PC: 0012, State: 0002
00> Registers: 
00>      r00:002c r01:0001 r02:0000 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: load r0, r2

00> PC: 0014, State: 0002
00> Registers: 
00>      r00:002c r01:0001 r02:0001 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: equ r2, r1

00> PC: 0016, State: 0003
00> Registers: 
00>      r00:002c r01:0001 r02:0001 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: br 4

00> PC: 001a, State: 0003
00> Registers: 
00>      r00:002c r01:0001 r02:0001 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: inc r1

00> PC: 001c, State: 0003
00> Registers: 
00>      r00:002c r01:0002 r02:0001 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: equ r1, r4

00> PC: 001e, State: 0002
00> Registers: 
00>      r00:002c r01:0002 r02:0001 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: br 10

00> PC: 0020, State: 0002
00> Registers: 
00>      r00:002c r01:0002 r02:0001 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: add r3, r0

00> PC: 0022, State: 0002
00> Registers: 
00>      r00:002e r01:0002 r02:0001 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: jr 240

00> PC: 0012, State: 0002
00> Registers: 
00>      r00:002e r01:0002 r02:0001 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: load r0, r2

00> PC: 0014, State: 0002
00> Registers: 
00>      r00:002e r01:0002 r02:0002 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: equ r2, r1

00> PC: 0016, State: 0003
00> Registers: 
00>      r00:002e r01:0002 r02:0002 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: br 4

00> PC: 001a, State: 0003
00> Registers: 
00>      r00:002e r01:0002 r02:0002 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: inc r1

00> PC: 001c, State: 0003
00> Registers: 
00>      r00:002e r01:0003 r02:0002 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: equ r1, r4

00> PC: 001e, State: 0002
00> Registers: 
00>      r00:002e r01:0003 r02:0002 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: br 10

00> PC: 0020, State: 0002
00> Registers: 
00>      r00:002e r01:0003 r02:0002 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: add r3, r0

00> PC: 0022, State: 0002
00> Registers: 
00>      r00:0030 r01:0003 r02:0002 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: jr 240

00> PC: 0012, State: 0002
00> Registers: 
00>      r00:0030 r01:0003 r02:0002 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: load r0, r2

00> PC: 0014, State: 0002
00> Registers: 
00>      r00:0030 r01:0003 r02:0003 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: equ r2, r1

00> PC: 0016, State: 0003
00> Registers: 
00>      r00:0030 r01:0003 r02:0003 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: br 4

00> PC: 001a, State: 0003
00> Registers: 
00>      r00:0030 r01:0003 r02:0003 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: inc r1

00> PC: 001c, State: 0003
00> Registers: 
00>      r00:0030 r01:0004 r02:0003 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: equ r1, r4

00> PC: 001e, State: 0003
00> Registers: 
00>      r00:0030 r01:0004 r02:0003 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: br 10

00> PC: 0028, State: 0003
00> Registers: 
00>      r00:0030 r01:0004 r02:0003 r03:0002 r04:0004 r05:0000 r06:0000 r07:0000
00>      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
00> Next instruction: (null) 

